WordPress/WordPressTest/BlogJetpackTests.swift

```diff
@@ -0,0 +1,264 @@
+import XCTest
+import CoreData
+import OHHTTPStubs
+@testable import WordPress
+
+class BlogJetpackTests: CoreDataTestCase {
+
+    // Properties
+
+    private var timeout: TimeInterval = 2.0
+    private var blog: Blog!
+    private var accountService: AccountService!
+    private var blogService: BlogService!
+
+    override func setUp() {
+        super.setUp()
+
+        // TODO: Account, Blog, testMainContext
+        blog = makeBlog()
+        accountService = .init(managedObjectContext: mainContext)
+        blogService = .init(managedObjectContext: mainContext)
+    }
+
+    override func tearDown() {
+        // TODO
+        blog = nil
+        accountService = nil
+        blogService = nil
+        HTTPStubs.removeAllStubs()
+
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func testJetpackInstalled() {
+        XCTAssertTrue(jetpackState.isInstalled)
+        blog.options = nil
+        XCTAssertNil(blog.jetpack)
+    }
+
+    func testJetpackVersion() {
+        XCTAssertEqual(jetpackState.version, "1.8.2")
+    }
+
+    func testJetpackSiteId() {
+        XCTAssertEqual(jetpackState.siteID?.intValue, 3)
+    }
+
+    func testJetpackUsername() {
+        XCTAssertNil(jetpackState.connectedUsername)
+    }
+
+    func testJetpackSetupDoesntReplaceDotcomAccount() {
+        var saveExpectation = expectation(forNotification: .NSManagedObjectContextDidSave, object: mainContext)
+        let account = accountService.createOrUpdateAccount(withUsername: "user", authToken: "token")
+        wait(for: [saveExpectation], timeout: timeout)
+        let defaultAccount = try? WPAccount.lookupDefaultWordPressComAccount(in: mainContext)
+
+        // the newly added account should be assigned as the default
+        XCTAssertEqual(account, defaultAccount)
```

Comment:
Would it make sense to break the second bit into a new test? Or do we need them to be chained?

Reply:
I'm not sure, actually. My aim was to migrate them to Swift, but keep the side effects as little as possible. So this is a good idea, but I would rather address this separately... Does that make sense? I can follow up on this as a tech debt later after we've shipped all the other stuff in the project.

Reply:
On a second look, for this use case I think this needs to be chained, yeah. This test makes sure that the second account creation through AccountService doesn't replace the default.

But you have a point there, the first assertion perhaps should be separated / copied into its own test, just to test the first assignment.